### ISSUE TITLE
Data tables performance improvements

### ISSUE THREAD

**Issue by vedina** [ideaconsult/Toxtree.js#179]
The display of search results for chemical structures needs to be optimised in order to become more responsive.  For a large set of results currently the screen stays empty and the data table appears only after the entire result is filled in.  I can see two ways of improvements:
* fill in the datatable as the results come in (this was done for older versions predating jToxKit )
* introduce [server side support](https://www.datatables.net/examples/data_sources/server_side.html) for datatables, filling in only the first page initially and perform queries on subsequent pages as the users browses the data.

@thejonan @gonzomir what are your preferences and thoughts?  Can we discuss it this week (skype)?

**Comment by thejonan** [comment]
I think, we've discussed that long ago with the conclusion that we need server-side paging support. Actually, I thought we already have it.

Option one looks tempting, because it seems like "abstracting the visualisation problem, to the visualisation part, i.e. UI", but in the end, it'll overwhelm the browser with too much data.

**Comment by vedina** [comment]
Yes, there is paging support.

**Comment by thejonan** [comment]
So, I'm missing some details then. Do we know what exactly is the slowest part? Could it be the server preparing the query?
I guess we can do some profiling and checking time consumption of each stage and request - time for request, time before data starts arriving, download time, etc.

**Comment by vedina** [comment]
Yes, we shall start with profiling. My guess at this moment is the bottleneck it is the data transferred , as I have seen the uncomfortable delay on saturated connections, and not on fast ones. But it could be combination of factors.
